src/components/panel/TextRenderer.tsx

```diff
@@ -149,22 +149,42 @@ const TextRenderer: FC<Props> = memo(({ htmlString }) => {
     return parser.parseFromString(`${htmlString}<span class="${END_CLASS}"></span>`, 'text/html')
   }, [htmlString])
 
-  const matchedAnnotationsMap = React.useMemo(() => {
+  React.useMemo(() => {
     if (!panelState.annotations) return {}
-    return panelState.annotations.reduce((acc, cur) => {
+    const result: MatchedAnnotationsMap = panelState.annotations.reduce((acc, cur) => {
       const matchedNodes = Array.from(parsedDom.body.querySelectorAll(cur.target[0].selector.value))
       if (matchedNodes.length > 0) {
-        acc[cur.id] = matchedNodes
+        const annotType = cur.body['x-content-type']
+        acc[cur.id] = { target: matchedNodes, filtered: fullAnnotationTypes[annotType] ?? true }
       }
       return acc
     }, {})
+
+    setMatchedAnnotationsMap(result)
+    return result
   }, [parsedDom, panelState.annotations])
 
   useEffect(() => {
-    const filteredAnnotations = panelState.annotations?.filter(a => matchedAnnotationsMap[a.id]) ?? []
-    setFilteredAnnotations(filteredAnnotations)
+    if (!panelState.annotations || panelState.annotations?.length === 0) return
+    // when switching to a new item, we extend our "full" annotationTypes
+    const newAnnotationTypes = getUpdatedAnnotationTypes(panelState.annotations, fullAnnotationTypes)
+    setFullAnnotationTypes(newAnnotationTypes)
   }, [matchedAnnotationsMap])
 
+  function getUpdatedAnnotationTypes(annotations: Annotation[], annotationTypes) {
```

Comment:
not a good function name, name should be descriptive for whats going on inside, "get updated" tells me it gets something updated from somewhere